charm/zookeeper/files/check_zookeeper.py

```diff
@@ -174,9 +176,23 @@ def analyze(self, opts, cluster_stats):
 
 class ZooKeeperServer(object):
 
-    def __init__(self, host='localhost', port='2181', timeout=1):
+    def __init__(self, host='localhost', port='2181', timeout=1,
+                 meta_file='/tmp/zk_check/meta'):
         self._address = (host, int(port))
         self._timeout = timeout
+        self._last_reset = datetime.utcnow()
+        self._meta_path = meta_file
+
+        directory = os.path.dirname(meta_file)
+        if not os.path.exists(directory):
+            os.makedirs(directory)
+
+        with open(meta_file, 'w+') as f:
+            buf = f.read()
+            if buf:
+                self._last_reset = datetime.utcfromtimestamp(float(buf))
```

Comment:
This should reset the timestamp file (assume no prior timestamp written) on `ValueError`. The file could be left empty if the process crashes or is killed after opening the file for writing but before anything is written.